### PR TITLE
[database]: Update the DPU database service configuration

### PR DESCRIPTION
**Pull request by Pterosaur**
#### Why I did it
According to the HLD: https://github.com/sonic-net/SONiC/pull/1534, update the DPU database configuration.
##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
1. Add the remote redis configuration in database_config.json.j2
2. Rename the database_global.json.j2
3. Determine the remote DB port according to the midplane IP
4. Get the DPU information from platform.json

#### How to verify it
Check them locally

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: dockers/docker-database/database_config.json.j2 [hunk 1]
```diff
@@ -102,23 +111,23 @@
         "DPU_APPL_DB" : {
             "id" : 15,
             "separator": ":",
-            "instance" : "redis",
+            "instance" : {% if include_remote_db%} "remote_redis" {% else %} "redis" {% endif %},
```
Comment (prgeor) [inline dockers/docker-database/database_config.json.j2:114]: @Pterosaur which design doc are you following? I am not sure, if the DPU_<Table Name> will be hosted on the NPU side or DPU side?

File: dockers/docker-database/docker-database-init.sh [hunk 2]
```diff
@@ -28,6 +28,19 @@ if [[ $DATABASE_TYPE == "dpudb" ]]; then
     redis_port=`expr 6381 + $DPU_ID`
 fi
 
+# TODO: Check the current platform is DPU
```
Comment (prgeor) [inline dockers/docker-database/docker-database-init.sh:31]: @Pterosaur can we keep the DPU port derivation based upon the midplane IP outside in a separate service that manages the mid-plane IP configuration

File: dockers/docker-database/docker-database-init.sh [hunk 3]
```diff
@@ -28,6 +28,19 @@ if [[ $DATABASE_TYPE == "dpudb" ]]; then
...
+if [[ $midplane_ip != "" ]]
+then
+    export DATABASE_TYPE="dpudb"
+    export REMOTE_DB_IP="169.254.200.254"
+    IFS=. read -r a b c d <<< "169.254.200.1"
+    midplane_base_ip_num=$((a * 256 ** 3 + b * 256 ** 2 + c * 256 + d))
+    # Determine the DB PORT from midplane IP
+    IFS=. read -r a b c d <<< $midplane_ip
+    midplane_ip_num=$((a * 256 ** 3 + b * 256 ** 2 + c * 256 + d))
+    export REMOTE_DB_PORT=$((6381 + $midplane_ip_num - $midplane_base_ip_num))
```
Comment (prgeor) [inline dockers/docker-database/docker-database-init.sh:42]: @Pterosaur you can get the ip from the "eth0-midplane" interface. Use the last digit (1 for DPU1, 2 for DPU 2, and so on)

REMOTE_DB_PORT = 6380 + <last digit of the IP>

File: dockers/docker-database/database_config.json.j2 (hunk 1 above)
Comment (Pterosaur) [reply dockers/docker-database/database_config.json.j2:114]: Add the design doc to the description and that has been reviewed by community.

File: dockers/docker-database/docker-database-init.sh (hunk 3 above)
Comment (Pterosaur) [reply dockers/docker-database/docker-database-init.sh]: Good suggestion, done.

**Review by ganglyu — APPROVED** [review]

**Comment by ganglyu** [comment]
For gnmi and other containers, the `/var/run/redis` directory is mounted by default, but the `/var/run/redisdpux` is not mounted. And then we can't access database configuration for dpu.

**Comment by Pterosaur** [comment]
> For gnmi and other containers, the `/var/run/redis` directory is mounted by default, but the `/var/run/redisdpux` is not mounted. And then we can't access database configuration for dpu.

Done

File: dockers/docker-database/database_config.json.j2 [hunk 4]
```diff
@@ -15,6 +16,14 @@
             "unix_socket_path": "/var/run/redis-chassis/redis_chassis.sock",
             "persistence_for_warm_boot" : "yes"
         }
+{% if include_remote_db %}
+        ,"remote_redis":{
```
Comment (qiluo-msft) [inline dockers/docker-database/database_config.json.j2:20]: >, [](http://example.com/codeflow?start=8&length=1)

The `,` convention will be weird. Would you be able to move it to previous line, I am ok with this j2 file mess up.

File: dockers/docker-database/database_config.json.j2 (hunk 1 above)
Comment (qiluo-msft) [inline dockers/docker-database/database_config.json.j2:114]: >% [](http://example.com/codeflow?start=48&length=1)

Do you want to add a blank before `%`?

Comment (Pterosaur) [reply dockers/docker-database/database_config.json.j2]: Done

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**